### ISSUE TITLE
Add `c:music_discs` in 1.21

### ISSUE THREAD

**Issue by TelepathicGrunt** [neoforged/NeoForge#993]
Mojang removed the tag. I think it was worth having for things like dropping a random music disc from a musical mob or spawning a random music disc in special chests or music disc input recipes
![image](https://github.com/neoforged/NeoForge/assets/40846040/c28ccb0b-6852-4ed6-ad05-0eb312315f3d)

https://github.com/FabricMC/fabric/issues/3790

**Comment by TelepathicGrunt** [comment]
Another idea:

`c:discs`
`c:discs/music`

But seems limited in use cases. Maybe just `c:music_discs` is good enough

**Comment by marchermans** [comment]
I don't think this is needed.
Given that something is a music disk based on a component being present, modeling this with a tag seems very counterproductive.

**Comment by TelepathicGrunt** [comment]
@marchermans counterpoint:

We aren't removing the food tags because food components exist. The tags are to mark categories and do groupings of items. Which is a very different use case than components which are specifically about functionality.

No music tag means a modder has to manually create a component ingredient for their recipes. It also means they need to make their own tag if they have a special boss that drops a random music disc on death (or manually drop the disc by code without loot table)

Lack of tag is also not friendly to datapackers who may want mod compat out of box as now they have to manually list out every modded music disc in their own tags since datapackers do not have component checking stuff modders can code in.

Basically, if I want a recipe to accept music disc *objects*, that's a tag for grouping that I want as my final product may be some decor with disc-like objects on it. I don't want *music playing* functionality items that could be attached to non-music-disc looking objects which would make my recipe all weird looking.

**Comment by TelepathicGrunt** [comment]
Basically, tags and component checking aren't mutually exclusive. Both can co-exist. I don't see why a music disc grouping tag should be yeeted in favor of components or vice versa. Can have a tag and also add component ingredients and the likes for modders.

**Comment by marchermans** [comment]
I thought about that for a while. 

But the question comes down to this: _What does the tag represent?_ in my eyes it represents an item which plays music when put in a Jukebox.

Based on that definition, you need to ask: Well when i want to list all items that match that criteria, what do they have in common? And the answer to that is that they have a specific component.

The problem with that and a tag is that you can add a component to any item (via command blocks for example) so it is easy to create an item that is not in the tag, but still a music disk.

So that makes me feel like that it does not make sense to have a tag for this. But to have modders check for the component.

**Comment by lukebemish** [comment]
The same could be said of the food component. The point of tags is that they can be used in all the places vanilla supports tags with no extra effort. In this case, a "music disc" is a record-looking thing of the same category as the vanilla music discs. There may be jukebox-playable things that aren't music discs, and I can even think of scenarios where there might be "music discs" that are not jukebox-playable. Convention tags aren't about "this item does XYZ", they're about "this item is in XYZ category"

**Comment by MehVahdJukaar** [comment]
So I'll add my bit to this. I have an item which plays music when put in jukeboxes. It is however not a music disc. In this case it would have the music disc component but not the musc disc tag as I would not want it to for example drop from creeper explosions or any other mod interactions as it looks very different from a normal disc since it's actually a food item
